### PR TITLE
Pass through SSSOM JSON-LD context when writing JSON

### LINKED ISSUE #477: `@context` missing the type definitions in version 0.4.0
The new release changed the `@context` when calling `to_json`, now only returning prefixes, instead of a complete definition of types in JSON-LD.

Expected `@context` returned in version 0.3.41

<details><summary>Click to view context version 0.3.41</summary>
<p>

```json
{
"@context":{
"dcterms":"http://purl.org/dc/terms/",
"linkml":"https://w3id.org/linkml/",
"oboInOwl":"http://www.geneontology.org/formats/oboInOwl#",
"owl":"http://www.w3.org/2002/07/owl#",
"pav":"http://purl.org/pav/",
"prov":"http://www.w3.org/ns/prov#",
"rdf":"http://www.w3.org/1999/02/22-rdf-syntax-ns#",
"rdfs":"http://www.w3.org/2000/01/rdf-schema#",
"semapv":"https://w3id.org/semapv/vocab/",
"skos":"http://www.w3.org/2004/02/skos/core#",
"sssom":"https://w3id.org/sssom/",
"@vocab":"https://w3id.org/sssom/",
"author_id":{
"@type":"@id",
"@id":"pav:authoredBy"
},
"comment":{
"@id":"rdfs:comment"
},
"confidence":{
"@type":"xsd:double"
},
"creator_id":{
"@type":"@id",
"@id":"dcterms:creator"
},
"curation_rule":{
"@type":"@id"
},
"documentation":{
"@type":"@id"
},
"homepage":{
"@type":"@id"
},
"imports":{
"@type":"@id"
},
"issue_tracker":{
"@type":"@id"
},
"issue_tracker_item":{
"@type":"@id"
},
"last_updated":{
"@type":"xsd:date"
},
"license":{
"@type":"@id",
"@id":"dcterms:license"
},
"mapping_cardinality":{
"@context":{
"@vocab":"@null",
"text":"skos:notation",
"description":"skos:prefLabel",
"meaning":"@id"
}
},
"mapping_date":{
"@type":"xsd:date",
"@id":"pav:authoredOn"
},
"mapping_justification":{
"@type":"@id"
},
"mapping_provider":{
"@type":"@id"
},
"mapping_registry_id":{
"@type":"@id"
},
"mapping_set_description":{
"@id":"dcterms:description"
},
"mapping_set_id":{
"@type":"@id"
},
"mapping_set_references":{
"@type":"@id"
},
"mapping_set_source":{
"@type":"@id",
"@id":"prov:wasDerivedFrom"
},
"mapping_set_title":{
"@id":"dcterms:title"
},
"mapping_set_version":{
"@id":"owl:versionInfo"
},
"mapping_source":{
"@type":"@id"
},
"mappings":{
"@type":"@id"
},
"mirror_from":{
"@type":"@id"
},
"object_id":{
"@type":"@id",
"@id":"owl:annotatedTarget"
},
"object_match_field":{
"@type":"@id"
},
"object_preprocessing":{
"@type":"@id"
},
"object_source":{
"@type":"@id"
},
"object_type":{
"@context":{
"@vocab":"@null",
"text":"skos:notation",
"description":"skos:prefLabel",
"meaning":"@id"
}
},
"predicate_id":{
"@type":"@id",
"@id":"owl:annotatedProperty"
},
"predicate_modifier":{
"@context":{
"@vocab":"@null",
"text":"skos:notation",
"description":"skos:prefLabel",
"meaning":"@id"
}
},
"predicate_type":{
"@context":{
"@vocab":"@null",
"text":"skos:notation",
"description":"skos:prefLabel",
"meaning":"@id"
}
},
"publication_date":{
"@type":"xsd:date",
"@id":"dcterms:created"
},
"registry_confidence":{
"@type":"xsd:double"
},
"reviewer_id":{
"@type":"@id"
},
"see_also":{
"@id":"rdfs:seeAlso"
},
"semantic_similarity_score":{
"@type":"xsd:double"
},
"subject_id":{
"@type":"@id",
"@id":"owl:annotatedSource"
},
"subject_match_field":{
"@type":"@id"
},
"subject_preprocessing":{
"@type":"@id"
},
"subject_source":{
"@type":"@id"
},
"subject_type":{
"@context":{
"@vocab":"@null",
"text":"skos:notation",
"description":"skos:prefLabel",
"meaning":"@id"
}
},
"Mapping":{
"@id":"owl:Axiom"
},
"HP":"http://purl.obolibrary.org/obo/HP_",
"MP":"http://purl.obolibrary.org/obo/MP_",
"orcid":"https://orcid.org/",
"obo":"http://purl.obolibrary.org/obo/",
}
}
```

</p>
</details> 

<details><summary>Click to view context in version 0.4.0</summary>
<p>

```json
{
"@context":{
"owl":"http://www.w3.org/2002/07/owl#",
"rdf":"http://www.w3.org/1999/02/22-rdf-syntax-ns#",
"rdfs":"http://www.w3.org/2000/01/rdf-schema#",
"semapv":"https://w3id.org/semapv/vocab/",
"skos":"http://www.w3.org/2004/02/skos/core#",
"sssom":"https://w3id.org/sssom/",
"HP":"http://purl.obolibrary.org/obo/HP_",
"MP":"http://purl.obolibrary.org/obo/MP_",
"obo":"http://purl.obolibrary.org/obo/",
"orcid":"https://orcid.org/",
}
}
```
</p>
</details> 

Is there another method to have the complete `@context` for JSON-LD file?

### PR DESCRIPTION
**Pull request by matentzn**
Fixes #477

### CODE REVIEW

File: src/sssom/writers.py [hunk 1]
```diff
@@ -436,7 +436,8 @@ def to_fhir_json(msdf: MappingSetDataFrame) -> Dict:
 def to_json(msdf: MappingSetDataFrame) -> JsonObj:
     """Convert a mapping set dataframe to a JSON object."""
     doc = to_mapping_set_document(msdf)
-    data = JSONDumper().dumps(doc.mapping_set, contexts={"@context": doc.prefix_map})
```
Comment (cthoyt) [inline src/sssom/writers.py:439]: I see right here there is directly an `@context`, what is the problem?

Comment (matentzn) [reply src/sssom/writers.py:439]: There is a lot of stuff missing in that context, pertaining to SSSOM internal elements. Look at what Anita posted how it was previously.

Comment (cthoyt) [reply src/sssom/writers.py:439]: fixed in eec2faa but we need a proper unit test

File: src/sssom/writers.py [hunk 2]
```diff
@@ -433,10 +435,29 @@ def to_fhir_json(msdf: MappingSetDataFrame) -> Dict:
     return json_obj
 
 
+def _update_sssom_context_with_prefixmap(converter: Converter) -> str:
+    """Prepare a JSON-LD context and dump to a string."""
+    context = _load_sssom_context()
+    for k, v in dict(converter.bimap).items():
+        if isinstance(v, str):
```
Comment (cthoyt) [inline src/sssom/writers.py:442]: unnecessary. bimap always has strings in its values

File: src/sssom/writers.py [hunk 3]
```diff
@@ -433,10 +435,29 @@ def to_fhir_json(msdf: MappingSetDataFrame) -> Dict:
...
+            if k not in context["@context"]:
```
Comment (cthoyt) [inline src/sssom/writers.py:443]: why do all of this checking? constructing the bimap and checking it doesn't override the built in prefix map is already done in a different place

Comment (matentzn) [reply src/sssom/writers.py]: You are right, I copy pasted what was there be for and didn't think that it is not needed anymore.

**Review by hrshdhgd — APPROVED** [review]

**Pull request merged**